lib/backend/k8s/k8s.go

```diff
@@ -210,6 +211,33 @@ func NewKubeClient(ca *apiconfig.CalicoAPIConfigSpec) (api.Client, error) {
 	return kubeClient, nil
 }
 
+// deduplicate removes any duplicated values and returns a new slice, keeping the order unchanged
+func deduplicate(s []string) []string {
+	encountered := map[string]bool{}
+	ret := make([]string, 0)
+	for i := range s {
```

Comment:
You can do
```
for _, v := range s {
   if encountered[v] {...
```
to directly iterate over the values instead of having to index each time.

Reply:
I think `v` would be a copy now instead of s[i], which would be a reference.. yeah it's weird for such a small func :smile: 

I just pirated this function from https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/loader.go#L635 but we can improve on it certainly

Reply:
`for _, v` is semantically a copy, sure, but in practice I'd expect zero difference here; string is a pointer-like type and the compiler is likely to spot that it doesn't need to actually copy.  Also not like this is hot-path critical code!

If you took that code from the k8s repo, please stick a copyright header on it as required by the Apache license.

Reply:
Added the copyright, I just left the function mostly intact